core/src/main/scala/caliban/wrappers/Wrapper.scala

```diff
@@ -56,7 +56,22 @@ object Wrapper {
    * Wrapper for the query validation stage.
    * Wraps a function from a `Document` to a `ZIO[R, ValidationError, ExecutionRequest]`.
    */
-  trait ValidationWrapper[-R] extends SimpleWrapper[R, ValidationError, ExecutionRequest, Document]
+  trait ValidationWrapper[-R] extends SimpleWrapper[R, ValidationError, ExecutionRequest, Document] { self =>
+
+    /**
+     * Returns a new wrapper which skips the [[wrap]] function if the query is an introspection query.
+     */
+    final def skipForIntrospection: ValidationWrapper[R] = new ValidationWrapper[R] {
+      override val priority: Int = self.priority
+
+      override def wrap[R1 <: R](
+        f: Document => ZIO[R1, ValidationError, ExecutionRequest]
+      ): Document => ZIO[R1, ValidationError, ExecutionRequest] =
+        (doc: Document) =>
+          if (doc.isIntrospection) f(doc)
+          else self.wrap(f)(doc)
+    }
```

Comment:
@ghostdogpr turns out we can't simply override `apply` and `|+|`. Do we think there might be cases where someone implemented custom logic in those methods?

For `apply`, I could get this to work, but looks rather weird so I left it out - let me know if you think we should add it:

```scala
override def apply[R1 <: R](that: GraphQL[R1]): GraphQL[R1] = super.apply(self.apply(that))
```

Reply:
It's probably okay